pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -970,3 +972,50 @@ message StatusResponse {
     // Status of the Runtime.
     RuntimeStatus status = 1;
 }
+
+message ImageFsInfoRequest {}
+
+// UInt64Value is the wrapper of uint64.
+message UInt64Value {
+    // The value.
+    uint64 value = 1;
+}
+
+// InodeInfo is the inode information of the images.
+message InodeInfo {
+    // Number of Inodes.
+    UInt64Value inodes = 7;
+    // Number of available Inodes (if known)
+    UInt64Value inodes_free = 8;
+}
+
+// FsInfo contains data about filesystem usage.
+message FsInfo {
+    // The block device name associated with the filesystem.
+    string device = 1;
+    // The root directory for the images.
+    string path = 2;
+    // CapacityBytes represents the total capacity (bytes) of the filesystems
+    // underlying storage.
+    UInt64Value capacity_bytes = 3;
+    // AvailableBytes represents the storage space available (bytes) for the
+    // filesystem.
+    UInt64Value available_bytes = 4;
+    // UsedBytes represents the bytes used for images on the filesystem.
+    // This may differ from the total bytes used on the filesystem and may not 
+    // equal CapacityBytes - AvailableBytes.
+    UInt64Value used_bytes = 5;
+    // InodesCapacity represents the total inodes in the filesystem.
+    UInt64Value inodes_capacity = 6;
+    // InodesAvailable represents the free inodes in the filesystem.
+    UInt64Value inodes_available = 7;
+    // InodesUsed represents the inodes used by the filesystem
+    // This may not equal InodesCapacity - InodesAvailable because this filesystem
```

Comment:
It could be shared between host and containers, or multiple containers.

Refer https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/volume.go#L75